state-sync/src/request_manager.rs

```diff
@@ -123,109 +111,105 @@ impl RequestManager {
         peer: PeerNetworkId,
         metadata: ConnectionMetadata,
     ) -> Result<(), Error> {
-        if !self.is_upstream_peer(&peer, metadata.origin) {
-            return Err(Error::PeerIsNotUpstream(
+        if !self.is_valid_state_sync_peer(&peer, metadata.origin) {
+            return Err(Error::InvalidStateSyncPeer(
                 peer.to_string(),
                 metadata.origin.to_string(),
             ));
         }
 
-        debug!(LogSchema::new(LogEntry::NewPeer)
+        info!(LogSchema::new(LogEntry::NewPeer)
             .peer(&peer)
-            .is_upstream_peer(true));
+            .is_valid_peer(true));
         counters::ACTIVE_UPSTREAM_PEERS
             .with_label_values(&[&peer.raw_network_id().to_string()])
             .inc();
 
-        if let Some(peer_info) = self.peers.get_mut(&peer) {
-            peer_info.is_alive = true;
-        } else {
-            let peer_score = if metadata.role == PeerRole::PreferredUpstream {
-                STARTING_SCORE_PREFERRED
-            } else {
-                STARTING_SCORE
-            };
-            self.peers.insert(peer, PeerInfo::new(true, peer_score));
+        match self.peer_scores.entry(peer) {
+            Occupied(occupied_entry) => {
+                warn!(LogSchema::new(LogEntry::NewPeerAlreadyExists).peer(occupied_entry.key()));
+            }
+            Vacant(vacant_entry) => {
+                let peer_score = if metadata.role == PeerRole::PreferredUpstream {
+                    STARTING_SCORE_PREFERRED
+                } else {
+                    STARTING_SCORE
+                };
+                vacant_entry.insert(peer_score);
+            }
         }
-        self.update_peer_selection_data();
 
         Ok(())
     }
 
-    pub fn disable_peer(
-        &mut self,
-        peer: &PeerNetworkId,
-        origin: ConnectionOrigin,
-    ) -> Result<(), Error> {
-        debug!(LogSchema::new(LogEntry::LostPeer)
-            .peer(&peer)
-            .is_upstream_peer(self.is_upstream_peer(&peer, origin)));
```

Comment:
Yeah, that was kinda silly, `is_upstream_peer`, when you know it is if it's stored.